### PR TITLE
luci-mod-admin-full: Update leds.lua to show all the possible ports

### PR DESCRIPTION
**Pull request by stephengeorgewest**
This will show all the values that can be applied as it matches /sys/class/leds/*/ports/
Signed-off-by: Stephen <stephen.george.west@gmail.com>

Possible fix for #2728

### CODE REVIEW

**Comment by feckert** [comment]
DCO was not successful.
Please add a new line in your commit before sign off.
And then make a push force.
I think this will pass the check.

**Comment by feckert** [comment]
@jow I want to give a hint that on openwrt 18.06 as I know have kernel version 4.14 and 4.9
So If we add this commit to 18.06 I hope this will not break the other targets who are on other kernel versions and maybe have other led triggers.

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua [hunk 1]
```diff
@@ -121,13 +121,7 @@ usbport.size = 1
...
 	local port, ports = nil, {}
 	for port in util.imatch(m.uci:get("system", section, "port")) do
-		local b, n = port:match("^usb(%d+)-port(%d+)$")
-		if not (b and n) then
-			b, n = port:match("^(%d+)-(%d+)$")
-		end
-		if b and n then
-			ports[#ports+1] = "usb%u-port%u" %{ tonumber(b), tonumber(n) }
-		end
+			ports[#ports+1] = port
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: Please fix the indentation level here (remove one tab)

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua [hunk 2]
```diff
@@ -144,11 +138,11 @@ for p in nixio.fs.glob("/sys/bus/usb/devices/[0-9]*/manufacturer") do
 	usbdev:value(id, "%s (%s - %s)" %{ id, mf, pr })
 end
 
-for p in nixio.fs.glob("/sys/bus/usb/devices/*/usb[0-9]*-port[0-9]*") do
-	local bus, port = p:match("usb(%d+)-port(%d+)")
-	if bus and port then
-		usbport:value("usb%u-port%u" %{ tonumber(bus), tonumber(port) },
-		              "Hub %u, Port %u" %{ tonumber(bus), tonumber(port) })
+for p in nixio.fs.glob("/sys/bus/usb/devices/*/*-port[0-9]*") do
+	local port = p:match("/sys/bus/usb/devices/.*/(.*-port%d+)$")
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: Please use `p:match("([^/]+-port%d+)$")` as pattern.

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua [hunk 3]
```diff
@@ -144,11 +138,11 @@ for p in nixio.fs.glob("/sys/bus/usb/devices/[0-9]*/manufacturer") do
...
+	if port then
+		usbport:value("%s" %{ port },
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: This interpolation is redundant, simply pass `port` as-is (`usbport:value(port, ...)`)

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua [hunk 4]
```diff
@@ -144,11 +138,11 @@ for p in nixio.fs.glob("/sys/bus/usb/devices/[0-9]*/manufacturer") do
...
+		              "%s %s" %{ port, nixio.fs.stat("%s/device" %{p}) ~= nil and "(device attached)" or ""})
```
Comment (jow-) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: Please add helper variable a few lines above assigning the device presence state and make the string translatable:

```
local attached = nixio.fs.stat("%s/device" % p) and ' (%s)' % translate('device attached') or ''
usbport:value(port, port .. attached)
```

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua (hunk 1 above)
Comment (stephengeorgewest) [reply modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: fixed

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua (hunk 2 above)
Comment (stephengeorgewest) [reply modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: done

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua (hunk 3 above)
Comment (stephengeorgewest) [reply modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: done

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua (hunk 4 above)
Comment (stephengeorgewest) [reply modules/luci-mod-admin-full/luasrc/model/cbi/admin_system/leds.lua]: done

**Comment by Neustradamus** [comment]
@feckert: It was integrated in master?

**Comment by stephengeorgewest** [comment]
I don't think it was, but it is probably a moot point since openwrt 18 is end of life and the logic has all moved to JS: applications\luci-app-ledtrig-usbport\htdocs\luci-static\resources\view\system\led-trigger\usbport.js

I upgraded the device to 21 a while ago, and checking now, I see that it lists 2 usb leds.
image
but there are no usb triggers (where before it listed the root usb port that was always connected).
image

So, yeah, this is probably a useless PR now.

**Pull request closed without merging**